File: internal/domain/visitor.go
package domain

type Visitor struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Email       string `json:"email"`
	Tower       string `json:"tower"`
	FlatNo      string `json:"flat_no"`
	AddedByRole string `json:"added_by_role"`
	Status      string `json:"status"`
	OwnerEmail  string `json:"owner_email"`
	OwnerID     string `json:"owner_id"`
	CreatedAt   int64  `json:"created_at"` 
}

File: src/models/visitor.py
import time
from pydantic import BaseModel, EmailStr, Field

from src.constants.visitor_enum import VisitorStatus
from src.constants.role_enum import UserRole


class Visitor(BaseModel):
    ID: str
    Name: str
    Email: EmailStr
    Tower: str
    FlatNo: str
    AddedByRole: UserRole
    Status: VisitorStatus
    OwnerEmail: EmailStr
    OwnerID: str
    CreatedAt: int
    @classmethod
    def from_ddb(cls, item: dict) -> "Visitor":
        return cls(
            ID=item["ID"],
            Name=item["Name"],
            Email=item["Email"],
            Tower=item["Tower"],
            FlatNo=item["FlatNo"],
            AddedByRole=UserRole(item["AddedByRole"]),
            Status=VisitorStatus(item["Status"]),
            OwnerEmail=item["OwnerEmail"],
            OwnerID=item["OwnerID"],
            CreatedAt=item["CreatedAt"],
        )

